lib/ui/src/components/sidebar/Search.tsx

```diff
@@ -1,145 +1,320 @@
+    display: 'none',
   },
 }));
 
-const CancelButton = styled.button<CancelButtonProps>(({ theme }) => ({
-  border: 0,
-  margin: 0,
-  padding: 4,
-  textDecoration: 'none',
+const FocusKey = styled.code(({ theme }) => ({
+  position: 'absolute',
+  top: 6,
+  right: 12,
+  width: 16,
+  height: 16,
+  zIndex: 1,
+  lineHeight: '17px',
+  textAlign: 'center',
+  fontSize: '11px',
+  background: 'rgba(0,0,0,0.1)',
+  color: theme.color.mediumdark,
+  borderRadius: 2,
+  userSelect: 'none',
+  pointerEvents: 'none',
+}));
 
-  background: theme.appBorderColor,
-  borderRadius: '1em',
+const ClearIcon = styled(Icons)(({ theme }) => ({
+  width: 16,
+  height: 16,
+  padding: 4,
+  position: 'absolute',
+  top: 6,
+  right: 12,
+  zIndex: 1,
+  background: 'rgba(0,0,0,0.1)',
+  borderRadius: 16,
+  color: theme.color.darker,
   cursor: 'pointer',
-  opacity: 1,
-  transition: 'all 150ms ease-out',
+}));
 
-  position: 'absolute',
-  top: '50%',
-  transform: 'translateY(-50%)',
-  right: 2,
+const FocusContainer = styled.div({ outline: 0 });
 
-  '> svg': {
-    display: 'block',
-    height: 8,
-    width: 8,
-    color: theme.input.color,
-    transition: 'all 150ms ease-out',
-  },
+export const Search: FunctionComponent<{
+  children: SearchChildrenFn;
+  dataset: CombinedDataset;
+  isLoading?: boolean;
+  lastViewed: Selection[];
+  clearLastViewed: () => void;
+  initialQuery?: string;
+}> = ({ children, dataset, isLoading = false, lastViewed, clearLastViewed, initialQuery = '' }) => {
+  const api = useStorybookApi();
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [inputPlaceholder, setPlaceholder] = useState('Find components');
+  const [allComponents, showAllComponents] = useState(false);
 
-  '&:hover': {
-    background: opacify(0.1, theme.appBorderColor),
-  },
-}));
+  const selectStory = useCallback(
+    (id: string, refId: string) => {
+      if (api) api.selectStory(id, undefined, { ref: refId !== DEFAULT_REF_ID && refId });
+      inputRef.current.blur();
+      showAllComponents(false);
+    },
+    [api, inputRef, showAllComponents, DEFAULT_REF_ID]
+  );
 
-const FilterForm = styled.form<FilterFormProps>(({ theme, focussed }) => ({
-  transition: 'all 150ms ease-out',
-  borderBottom: '1px solid transparent',
-  borderBottomColor: focussed
-    ? opacify(0.3, theme.appBorderColor)
-    : opacify(0.1, theme.appBorderColor),
-  outline: 0,
-  position: 'relative',
-  color: theme.input.color,
-
-  input: {
-    color: theme.input.color,
-    fontSize: theme.typography.size.s2 - 1,
-    lineHeight: '20px',
-    paddingTop: 2,
-    paddingBottom: 2,
-    paddingLeft: 20,
-    paddingRight: 20,
-  },
+  useEffect(() => {
+    const focusSearch = (event: KeyboardEvent) => {
+      if (isLoading || !inputRef.current) return;
+      if (event.shiftKey || event.metaKey || event.ctrlKey || event.altKey) return;
+      if (event.key === '/' && inputRef.current !== document.activeElement) {
```

Comment:
There's a disable shortcut option that is not taken into account here, should it be ? (and hide shortcut in the meantime)

Reply:
Fixed. Good call!